executor/mem_reader.go

```diff
@@ -357,6 +350,40 @@ func iterTxnMemBuffer(ctx sessionctx.Context, kvRanges []kv.KeyRange, fn process
 	return nil
 }
 
+func getMemIter(tblID int64, ctx sessionctx.Context, iter kv.Iterator, rg kv.KeyRange) (kv.Iterator, error) {
+	var snapCacheIter kv.Iterator
+	tempTableData := ctx.GetSessionVars().TemporaryTableData
+	if tempTableData != nil {
+		snapIter, err := tempTableData.Iter(rg.StartKey, rg.EndKey)
+		if err != nil {
+			return nil, err
+		}
+		snapCacheIter = snapIter
+	}
+	cond := ctx.GetSessionVars().StmtCtx.GetCacheTableReadCondition(tblID)
+	if cond {
+		tbl, ok := domain.GetDomain(ctx).InfoSchema().TableByID(tblID)
```

Comment:
`ctx.GetInfoSchema()` only can get a SchemaMetaVersion() int64

Reply:
I don't think it can be reason. `ctx.GetInfoSchema()` returns `SchemaMetaVersion` is a tricky to avoid cycle depency, it can be recognized as `infoschema.Infoschema` in fact. `domain.GetDomain(ctx).InfoSchema()` returns the latest infoschema in the current instance, but `ctx.GetInfoSchema()` returns the infoschema you should use in current session. They are not always the same.